ublox/src/parser.rs

```diff
@@ -414,55 +427,71 @@ impl UbxChecksumCalc {
     }
 }
 
+#[derive(Debug, PartialEq, Eq)]
+enum NextSync {
+    Ubx(usize),
+    Rtcm(usize),
+    None,
+}
+
+#[derive(Debug)]
+pub enum AnyPacketRef<'a> {
+    Ubx(PacketRef<'a>),
+    Rtcm(RtcmPacketRef<'a>),
+}
+
 /// Iterator over data stored in `Parser` buffer
-pub struct ParserIter<'a, T: UnderlyingBuffer> {
+pub struct UbxParserIter<'a, T: UnderlyingBuffer> {
     buf: DualBuffer<'a, T>,
 }
 
-impl<T: UnderlyingBuffer> ParserIter<'_, T> {
-    fn find_sync(&self) -> Option<usize> {
-        (0..self.buf.len()).find(|&i| self.buf[i] == SYNC_CHAR_1)
-    }
-
-    fn extract_packet(&mut self, pack_len: usize) -> Option<Result<PacketRef, ParserError>> {
-        if !self.buf.can_drain_and_take(6, pack_len + 2) {
-            if self.buf.potential_lost_bytes() > 0 {
-                // We ran out of space, drop this packet and move on
-                self.buf.drain(2);
-                return Some(Err(ParserError::OutOfMemory {
-                    required_size: pack_len + 2,
-                }));
-            }
-            return None;
-        }
-        let mut checksummer = UbxChecksumCalc::new();
-        let (a, b) = self.buf.peek_raw(2..(4 + pack_len + 2));
-        checksummer.update(a);
-        checksummer.update(b);
-        let (ck_a, ck_b) = checksummer.result();
-
-        let (expect_ck_a, expect_ck_b) = (self.buf[6 + pack_len], self.buf[6 + pack_len + 1]);
-        if (ck_a, ck_b) != (expect_ck_a, expect_ck_b) {
-            self.buf.drain(2);
-            return Some(Err(ParserError::InvalidChecksum {
-                expect: u16::from_le_bytes([expect_ck_a, expect_ck_b]),
-                got: u16::from_le_bytes([ck_a, ck_b]),
+fn extract_packet_ubx<'b, T: UnderlyingBuffer>(
+    buf: &'b mut DualBuffer<'_, T>,
+    pack_len: usize,
+) -> Option<Result<PacketRef<'b>, ParserError>> {
+    if !buf.can_drain_and_take(6, pack_len + 2) {
```

Comment:
Is it possible to name these magic numbers?

Reply:
I totally agree, but considering the workload and that it is out of scope of the proposal, it's fine if we deal with those in separate PR (these are in the UbxParser after all, which already exists)

Reply:
#126